llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4546,6 +4546,12 @@ static bool willGenerateVectors(VPlan &Plan, ElementCount VF,
   return false;
 }
 
+static bool hasReplicatorRegion(VPlan &Plan) {
+  return any_of(VPBlockUtils::blocksOnly<VPRegionBlock>(
+                    vp_depth_first_deep(Plan.getEntry())),
```

Comment:
Ah sorry, it would need to be starting at the entry of the vector loop region I think, something like

```suggestion
                  vp_depth_first_shallow(Plan.getVectorLoopRegion()->getEntry())),
```

Reply:
It looks like that does work.